oauth2/oauth2-jose/src/test/java/org/springframework/security/oauth2/jose/TestKeys.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.jose;
+
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * @author Joe Grandja
+ * @since 5.2
+ */
+public class TestKeys {
+	public static final SecretKey DEFAULT_SECRET_KEY;
+
+	static {
+		try {
+			DEFAULT_SECRET_KEY = secretKey();
+		} catch (NoSuchAlgorithmException ex) {
+			throw new RuntimeException(ex);
+		}
+	}
+
+	public static SecretKey secretKey() throws NoSuchAlgorithmException {
+		KeyGenerator keyGenerator = KeyGenerator.getInstance("AES");
```

Comment:
Is there any reason to have this be generated? It would be nice if it were a static value, e.g.

```suggestion
		return new SecretKeySpec(Base64.getDecoder().decode("Ky8xmu1fg/OqVlUr9PRKhutauHvuj0rLHExRk+5XkSU="), "AES")
```

Reply:
There is no particular reason. I feel `secretKey()` can be useful as some tests may require their own secret. Is there a benefit to having a static value other than the one-time optimization at class load?

Reply:
Having a "pre-generated" one is faster. To be clear, I think `secretKey()`, should stay, it should just use load a pre-generated key instead of a live-generated key.

Also, taking a look at a few other tests, it is fairly common to use a pre-generated value, for example, `NimbusReactiveJwtDecoderTests#decodeWhenRSAPublicKeyThenSuccess`, and the resource server tests.

Reply:
> Having a "pre-generated" one is faster

True, but the generated key is done once at class load so I don't think this will make the tests run any much faster.

Reply:
I'd prefer to have this be a static value too. This is not only faster, but it ensures the tests are consistent and predictable.

Reply:
Looking through the tests, I don't see this method being used. Let's remove it, adding it back in when there is a clear need for a test to have a dynamically-generated secret key.